src/settings.c

```diff
@@ -88,12 +88,12 @@ void StartupSettings(void)
 }
 
 /** Update a string setting. */
-void SetPathString(char **dest, const char *src)
+void SetPathString(char *dest, const char *src)
 {
-   if(*dest) {
-      Release(*dest);
+   if(dest) {
+      Release(dest);
    }
-   *dest = CopyString(src);
+   dest = CopyString(src);
```

Comment:
I'm not exactly sure what the goal is, but I don't think this will work. This will release the memory used by the caller, allocate new memory, and then not pass it to the caller.

Reply:
cppcheck has given warning.